qiskit/primitives/base_sampler.py

```diff
@@ -0,0 +1,195 @@
+accessed within "with" context
+and the objects are called with parameter values and run options
+(e.g., ``shots`` or number of shots).
+
+Here is an example of how sampler is used.
+
+.. code-block:: python
+
+    from qiskit import QuantumCircuit
+    from qiskit.circuit.library import RealAmplitudes
+
+    bell = QuantumCircuit(2, 2)
+    bell.h(0)
+    bell.cx(0, 1)
+    bell.measure(0, 0)
+    bell.measure(1, 1)
+
+    # executes a Bell circuit
+    with Sampler(circuits=[bell], parameters=[[]]) as sampler:
+        result = sampler([[]])
+        print([q.binary_probabilities() for q in result.quasi_dists])
+
+    # executes three Bell circuits
+    with Sampler(circuits=[bell]*3) as sampler:
+        result = sampler([[]]*3)
+        print([q.binary_probabilities() for q in result.quasi_dists])
+
+    # parametrized circuit
+    pqc = QuantumCircuit(2, 2)
+    pqc.compose(RealAmplitudes(num_qubits=2, reps=2), inplace=True)
+    pqc.measure(0, 0)
+    pqc.measure(1, 1)
+
+    theta1 = [0, 1, 1, 2, 3, 5]
+    theta2 = [1, 2, 3, 4, 5, 6]
+
+    with Sampler(circuits=[pqc], parameters=[pcq.parameters]) as sampler:
+        result1 = sampler([theta1, theta2], [0, 0])
+
+        # result of pqc(theta1)
+        print([q.binary_probabilities() for q in result[0].quasi_dists])
+
+        # result of pqc(theta2)
+        print([q.binary_probabilities() for q in result[1].quasi_dists])
+
+"""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Callable, Sequence
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.circuit.parametertable import ParameterView
+from qiskit.exceptions import QiskitError
+
+from .sampler_result import SamplerResult
+
+
+class BaseSampler(ABC):
+    """Sampler base class
+
+    Base class of Sampler that calculates quasi-probabilities of bitstrings from quantum circuits.
+    """
+
+    def __init__(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameters: Sequence[Sequence[Parameter]] | None = None,
+    ):
+        """
+        Args:
+            circuits (list[QuantumCircuit]): quantum circuits to be executed
+            parameters (list[list[Parameter]]): parameters of quantum circuits
+                Defaults to `[circ.parameters for circ in circuits]`
+
+        Raises:
+            QiskitError: for mismatch of circuits and parameters list.
+        """
+        self._circuits = tuple(circuits)
+        if parameters is None:
+            self._parameters = tuple(circ.parameters for circ in circuits)
+        else:
+            if len(parameters) != len(circuits):
+                raise QiskitError(
+                    f"Different number of parameters ({len(parameters)} and circuits ({len(circuits)}"
+                )
+            self._parameters = tuple(ParameterView(par) for par in parameters)
+
+    def __call__(
+        self,
+        circuits: Sequence[int],
+        parameters: Sequence[Sequence[float]],
```

Comment:
@levbishop `grouping` comes after `parameters` in `Estimator.run`. On the other hand, `circuits` comes before `parameters` in `Sampler.run`. I feel they are not consistent. Do you plan to change the order?

Reply:
Yes i noticed it and will change so that its same order everywhere

Reply:
I think most sense is to be (circuit, [observable], params) everywhere both in the class init and also in the instance __call__ so

Reply:
Thanks. It sounds consistent.

Reply:
If we are going to enforce this order, I think it would be better to drop the tuples and Group and just use `__call__(circuits, parameters)` for the Sampler and `__call__(circuits, observables, parameters)` for the Estimator.

Reply:
Actually this is a good idea and a nice additional simplification. I agree